llama_index/query_engine/sql_vector_query_engine.py

```diff
@@ -0,0 +1,311 @@
+    "SQL query: SELECT city_name FROM cities WHERE continent = 'North America' LIMIT 3\n"
+    "SQL response: The cities in North America are New York, San Francisco, and Toronto.\n"
+    "New question: What sports are played in New York, San Francisco, and Toronto?\n\n"
+    "Original question: What is the city with the highest population?\n"
+    "SQL query: SELECT city, population FROM cities ORDER BY population DESC LIMIT 1\n"
+    "SQL response: The city with the highest population is New York City.\n"
+    "New question: None\n\n"
+    "Original question: What countries are the top 3 ATP players from?\n"
+    "SQL query: SELECT country FROM players WHERE rank <= 3\n"
+    "SQL response: The top 3 ATP players are from Serbia, Russia, and Spain.\n"
+    "New question: None\n\n"
+    "Original question: {query_str}\n"
+    "SQL query: {sql_query_str}\n"
+    "SQL response: {sql_response_str}\n"
+    "New question: "
+)
+DEFAULT_SQL_AUGMENT_TRANSFORM_PROMPT = Prompt(DEFAULT_SQL_AUGMENT_TRANSFORM_PROMPT_TMPL)
+
+
+def _default_check_stop(query_bundle: QueryBundle) -> bool:
+    """Default check stop function."""
+    return query_bundle.query_str.lower() == "none"
+
+
+def _format_sql_query(sql_query: str) -> str:
+    """Format SQL query."""
+    return sql_query.replace("\n", " ").replace("\t", " ")
+
+
+class SQLAugmentQueryTransform(BaseQueryTransform):
+    """SQL Augment Query Transform.
+
+    This query transform will transform the query into a more specific query
+    after augmenting with SQL results.
+
+    Args:
+        llm_predictor (LLMPredictor): LLM predictor to use for query transformation.
+    """
+
+    def __init__(
+        self,
+        llm_predictor: Optional[BaseLLMPredictor] = None,
+        sql_augment_transform_prompt: Optional[Prompt] = None,
+        check_stop_parser: Optional[Callable[[str], bool]] = None,
+    ) -> None:
+        """Initialize params."""
+        self._llm_predictor = llm_predictor or LLMPredictor()
+
+        self._sql_augment_transform_prompt = (
+            sql_augment_transform_prompt or DEFAULT_SQL_AUGMENT_TRANSFORM_PROMPT
+        )
+        self._check_stop_parser = check_stop_parser or _default_check_stop
+
+    def _run(self, query_bundle: QueryBundle, extra_info: Dict) -> QueryBundle:
+        """Run query transform."""
+        query_str = query_bundle.query_str
+        sql_query = extra_info["sql_query"]
+        sql_query_response = extra_info["sql_query_response"]
+        new_query_str, formatted_prompt = self._llm_predictor.predict(
+            self._sql_augment_transform_prompt,
+            query_str=query_str,
+            sql_query_str=sql_query,
+            sql_response_str=sql_query_response,
+        )
+        return QueryBundle(
+            new_query_str, custom_embedding_strs=query_bundle.custom_embedding_strs
+        )
+
+    def check_stop(self, query_bundle: QueryBundle) -> bool:
```

Comment:
I don't get why this is a method in `SQLAugmentQueryTransform`

Reply:
mostly because the query transform itself owns the prompt that can decide whether to generate a new query or stop after the sql response